docs/docs/contributing.md

```diff
@@ -53,35 +53,73 @@ number of relatively approachable issues with the label
-  you'd like to resolve potential merge conflicts. The idea behind is that we
-  don't want an arbitrary number of commits for one pull request, but exactly
-  one commit. This commit should be easy to merge onto master, therefore we
-  ask you to rebase to master.
-  
-- Please start your commit message with "Fixes #1234 - " where #1234 is the github issue number
-  that your pull request relates to. Github will automatically link this PR to the issue and make it more
-  visible to others.
+1. A GitHub pull request is the preferred way of submitting code changes.
+
+2. If a JIRA issue doesn't already exist, please open one for the proposed change in behavior (see the section above,
+   "JIRA Issues").
+
+3. Open up a discussion about the JIRA ticket (ideally, in the comments section, but, also in the
+   [#marathon channel](https://mesos.slack.com/messages/C1L7D22KY/). Seek to get buy-off and consensus on the direction
+   before making the change.
+
+4. Please rebase your pull requests on top of the current master, as needed, to integrate upstream changes and resolve
+   conflicts. You can squash your commits if it helps resolve potential merge conflicts. When your PR is merged, it will
+   be squashed to a single commit.
+
+5. If you are targeting to get something fixed for an older release of Marathon, fix it on master, first. If it is
+   already fixed in master, first, find out why. See our "Backporting Policy" above.
 
-- Any changes in the public API or behavior must be reflected in the project
-  documentation.
+6. Please include in your commit message the line "JIRA Issues: MARATHON-1234", where MARATHON-1234 is the JIRA issue
+   mentioned in item 2 above. This helps reviewers better understand the context of the change.
 
-- Any changes in the public API or behavior must be reflected in the changelog.md.
+7. If you change modifies the public API or behavior, then the project documentation must be updated (in the same pull
+   request). Further, notes about the change should be specified in changelog.md.
 
-- Pull requests should include appropriate additions to the unit test suite.
+8. Pull requests should include appropriate additions to the unit test suite (see "Test Guidelines", below). If the
+   change is a bugfix, then the added tests must fail without the fix as a safeguard against future regressions.
 
-- If the change is a bugfix, then the added tests must fail without the patch
-  as a safeguard against future regressions.
+9. Compile your code prior to committing. We would like the result of our automatic code formatters to be included in
+   the commit as to not produce a dirty work tree after fresh checkout and first compile.
 
-- Run all tests via the supplied `./bin/run-tests.sh` script (requires docker).
+10. Run, at the very least, all unit tests (`sbt test`). Integration tests can also be run using the supplied
+    `./bin/run-tests.sh` script (requires docker).
```

Comment:
Let's put something about `mvm.sh` here. It really works great.

Reply:
It's what we had previously. It's a good point about us not personally using it.

I prefer to use the version of Mesos I have installed locally on my laptop, rather than Docker. It's more efficient. I think there's a tradeoff between initial setup and recurring run time.

Reply:
Since no one in the team seems to use `./bin/run-tests.sh` we probably should not advocate it. I'd rather ask them to use `mvm.sh` as Nikita suggests. This is used by a few team members and has seen patches. Let's actually get rid off `run-tests`.